src/benchmark/listeners.ts

```diff
@@ -0,0 +1,97 @@
+import fibonacci from './cases/fibonacci';
+import newtonsMethod from './cases/newtons_method';
+import primeCounter from './cases/prime_counter';
+import quickSort from './cases/quicksort';
+import sumDoubles from './cases/sum_doubles';
+import sumIntegers from './cases/sum_integers';
+import Measurement from './Measurement';
+
+const algorithms = new Map<string, () => void>([
+    ['Fibonacci', fibonacci],
+    ['Newtons Method', newtonsMethod],
+    ['Prime Counter', primeCounter],
+    ['Quick Sort', quickSort],
+    ['Sum Doubles', sumDoubles],
+    ['Sum Integers', sumIntegers],
+]);
+
+function sum(value1: number, value2: number): number {
+    return value1 + value2;
+}
+
+function mean(values: number[]): number {
+    return values.reduce(sum, 0) / values.length;
+}
+
+function median(values: number[]): number {
+    if (values.length === 0) {
+        return 0;
+    }
+
+    if (values.length === 1) {
+        return values[0];
+    }
+
+    values.sort((a, b) => a - b);
+
+    const half = Math.floor(values.length / 2);
+
+    if (values.length % 2 === 0) {
+        return values[half];
+    } else {
+        return (values[half - 1] + values[half]) / 2;
+    }
+}
+
+function variance(values: number[]): number {
+    const m = mean(values);
```

Comment:
Rename the variable `m`

Reply:
The problem is, a variable can not be called the same name as a function. Therefore mean is not an option.